lib/rpc/handler/FallbackHandler.ts

```diff
@@ -0,0 +1,97 @@
+import { metricScope, MetricsLogger } from 'aws-embedded-metrics'
+import { APIGatewayProxyResult } from 'aws-lambda'
+import { AWSMetricsLogger } from '../../handlers/router-entities/aws-metrics-logger'
+
+interface AlarmEvent {
+  alarmName: string
+  state: string
+  previousState: string
+  providerId: string
+  reason: string
+}
+
+export class FallbackHandler {
+  private healthStateRepository: ProviderHealthStateRepository
+  private log: Logger
+
+  constructor(log: Logger) {
+    const dbTableName = process.env.PROVIDER_HEALTH_STATE_DB_TABLE_NAME
+    if (!dbTableName) {
+      throw new Error('Missing DB_TABLE_NAME in env var')
+    }
+    this.healthStateRepository = new ProviderHealthStateDynamoDbRepository(dbTableName, log)
+    this.log = log
+  }
+
+  private buildHandler() {
+    return metricScope((metricsLogger: MetricsLogger) => async (event: object): Promise<APIGatewayProxyResult> => {
+      metricsLogger.setNamespace('Uniswap')
+      metricsLogger.setDimensions({ Service: 'RoutingAPI' })
+      const metric = new AWSMetricsLogger(metricsLogger)
+
+      const alarmEvent = this.readAlarmEvent(event)
+      this.log.debug({ alarmEvent }, 'Parsed alarmEvent')
+
+      if (
+        (alarmEvent.previousState === 'OK' || alarmEvent.previousState === 'INSUFFICIENT_DATA') &&
+        alarmEvent.state === 'ALARM'
+      ) {
+        metric.putMetric(`RPC_GATEWAY_FALLBACK_${alarmEvent.providerId}_INTO_UNHEALTHY`, 1, MetricLoggerUnit.Count)
+        this.log.error(
+          `${alarmEvent.providerId} becomes UNHEALTHY due to ${alarmEvent.previousState}=>ALARM in ${alarmEvent.alarmName}`
+        )
+        await this.healthStateRepository.write(alarmEvent.providerId, ProviderHealthState.UNHEALTHY)
+      } else if (
+        alarmEvent.previousState === 'ALARM' &&
+        (alarmEvent.state === 'OK' || alarmEvent.state === 'INSUFFICIENT_DATA')
+      ) {
+        metric.putMetric(`RPC_GATEWAY_FALLBACK_${alarmEvent.providerId}_INTO_HEALTHY`, 1, MetricLoggerUnit.Count)
+        this.log.error(
+          `${alarmEvent.providerId} becomes HEALTHY due to ALARM=>${alarmEvent.state} in ${alarmEvent.alarmName}`
+        )
+        await this.healthStateRepository.write(alarmEvent.providerId, ProviderHealthState.HEALTHY)
+      }
+
+      return {
+        statusCode: 200,
+        body: '',
+      }
+    })
+  }
+
+  get handler() {
+    return async (event: object): Promise<APIGatewayProxyResult> => {
+      const handler = this.buildHandler()
+      this.log.debug({ event }, 'Received event object')
+      const response = await handler(event)
+      this.log.debug({ response }, 'Response of fallback handler')
+      return response
+    }
+  }
+
+  readAlarmEvent(event: any): AlarmEvent {
+    // Example alarm name: "RoutingAPI-RpcGateway-ErrorRateAlarm-ChainId-56-Provider-QUIKNODE"
+    const alarmName: string = event.alarmData.alarmName
+    const alarmNameTokens = alarmName.split('-')
```

Comment:
this works but it is a little brittle: is there any way we could use AWS tags instead ?

Reply:
Unfortunately we have never used it in all of the metrics created from this repo. I can add a TODO here about it, but it's kind of out of the scope of this PR.

Reply:
fair enough